### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by cscetbon** [kubernetes-sigs/controller-tools#624]
Using controller-ben v0.6.2 I get a panic error when trying to generate the crd for https://github.com/Orange-OpenSource/casskop/tree/fix-crds/multi-casskop

```
controller-gen --version
Version: v0.6.2
make controller-gen
make[1]: Entering directory '/home/circleci/*******/multi-*******'
Makefile:102: warning: overriding recipe for target 'build'
../shared.mk:127: warning: ignoring old recipe for target 'build'
make[1]: Nothing to be done for 'controller-gen'.
make[1]: Leaving directory '/home/circleci/*******/multi-*******'
/go/bin/controller-gen crd paths=./pkg/apis/... output:dir=./deploy/crds schemapatch:manifests=./deploy/crds
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xa15efb]

goroutine 1 [running]:
sigs.k8s.io/controller-tools/pkg/crd.schemaWalker.walkSchema({{0xcdc240, 0xc004cac1e0}}, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-tools@v0.6.2/pkg/crd/schema_visitor.go:63 +0xbb
sigs.k8s.io/controller-tools/pkg/crd.EditSchema(...)
	/go/pkg/mod/sigs.k8s.io/controller-tools@v0.6.2/pkg/crd/schema_visitor.go:42
sigs.k8s.io/controller-tools/pkg/crd.FlattenEmbedded(0xc0021048d0, {0xcdc280, 0xc0000ede00})
	/go/pkg/mod/sigs.k8s.io/controller-tools@v0.6.2/pkg/crd/flatten.go:210 +0x7b
sigs.k8s.io/controller-tools/pkg/crd.(*Parser).NeedFlattenedSchemaFor(0xc003649ce0, {0xc0000ede00, {0xc0004fa070, 0x2}})
	/go/pkg/mod/sigs.k8s.io/controller-tools@v0.6.2/pkg/crd/parser.go:204 +0xe8
sigs.k8s.io/controller-tools/pkg/schemapatcher.Generator.Generate({{0xc0001c293a, 0xb43d80}, 0x0, 0x0}, 0xc001c2f7c0)
	/go/pkg/mod/sigs.k8s.io/controller-tools@v0.6.2/pkg/schemapatcher/gen.go:143 +0x9e9
sigs.k8s.io/controller-tools/pkg/genall.(*Runtime).Run(0xc00024ca00)
	/go/pkg/mod/sigs.k8s.io/controller-tools@v0.6.2/pkg/genall/genall.go:207 +0x206
main.main.func1(0xc0002b8500, {0xc00028dc00, 0x4, 0x4})
	/go/pkg/mod/sigs.k8s.io/controller-tools@v0.6.2/cmd/controller-gen/main.go:176 +0x6a
github.com/spf13/cobra.(*Command).execute(0xc0002b8500, {0xc00019e010, 0x4, 0x4})
	/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002b8500)
	/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
main.main()
	/go/pkg/mod/sigs.k8s.io/controller-tools@v0.6.2/cmd/controller-gen/main.go:200 +0x2c6
make: *** [../shared.mk:122: generate] Error 2
```

I'm trying to replace `operator-sdk generate k8s` by using `controller-gen` but hit this error.

**Comment by cscetbon** [comment]
same with version 0.7.0

**Comment by abayer** [comment]
I hit this same panic/stack trace, though in a bit of a different scenario than I saw in your repo - in Tekton Pipelines, we have `pkg/apis/pipeline/{v1alpha1,v1beta1}, `pkg/apis/resource/v1alpha1`, and `pkg/apis/run/v1alpha1`, all with the same `tekton.dev` group. I did some debugging and discovered that it would fail trying to find a kind from, say, `pkg/apis/run/v1alpha1` in the package `pkg/apis/resource/v1alpha1`. I've got a bare fix up at https://github.com/abayer/controller-tools/commit/c8009f3483b75a66454a5b0ad89159ed40413c80, and will try to open a PR once I've gotten a test case set up.
